### PR TITLE
Refactor api.New() to take options for optional config

### PR DESCRIPTION
**Pull request by metalmatze**
/cc @lilic 

Sadly the prefix isn't able to just be done on the outside, as the current routers don't support routing. For that I'd suggest using go-chi/chi in the future.
Still a lot cleaner already.

### CODE REVIEW

**Comment by brancz** [comment]
Looks like some e2e tests are failing. Otherwise lgtm!

File: api/api.go [hunk 1]
```diff
@@ -78,48 +78,91 @@ type API struct {
...
-		db:                db,
-		reloadCh:          reloadCh,
-		maxMergeBatchSize: maxMergeBatchSize,
-		targets:           targets,
+	a := &API{
+		logger:   logger,
+		registry: registry,
+		prefix:   "/api/v1/",
+		reloadCh: make(chan struct{}),
 		globalURLOptions: GlobalURLOptions{ // TODO pass into from flags
```
Comment (lilic) [inline api/api.go:100]: Side note: Ah maybe this could be a nice first issue for someone to do, what do you if we open an issue for this?

**Review by lilic — COMMENTED** [review]
Nice!

**Review by brancz — APPROVED** [review]
Very nice!

**Pull request merged**